[add] floating-item smooth-scroll

sae-github opened the pull request:
#  floating-item / smooth-scroll 
## [CodeSandBox](https://codesandbox.io/s/elated-banzai-ogmeq?file=/index.html)
### Specification Description
-Click on the navigation link to scroll to the target section.
-The header will follow. Therefore, I adjusted the header so that it would not be covered when scrolling.
-When the window scrolls more than 100px, a "Back to Top" button will appear, which can be clicked to return to the top.
### Concerns and areas to focus on
-I had trouble with the variable names...
If you have any advice, I would appreciate it.

kenmori commented:
ok, your code look like good.
I suggest you to improve for performance.


- Is it possible to assign eventListener to all elements instead of assigning them to the parent element (ul) to determine where the event was clicked and scroll?
- Can you use intersectionObserver to return to Top? your code costs performance to judge for each scroll

sae-github commented:
Thanks for the review.
I was able to gain new knowledge.

> Is it possible to assign eventListener to all elements instead of assigning them to the parent element (ul) to determine where the event was clicked and scroll?

Fixed it.
I want you to tell me one thing.
Why is it better this way?
Is it because the elements to be acquired are more limited?

> Can you use intersectionObserver to return to Top? your code costs performance to judge for each scroll

Fixed it.

kenmori commented on src/index.js:
```diff
@@ -0,0 +1,38 @@
+  })
+}
+const observer = new IntersectionObserver(callback);
+observer.observe(document.getElementById('section3'));
```
> Why is it better this way?

Scroll events are calculated each time the user scrolls.
It is every 1px. 
If there is a height of 1000px, 
it will be executed 1000 times.
This is poor performance.

IntersectionObserver fires when the element appears on the screen, so the number of times it fires is different. In other words, the performance is good.

good,
Looking at your sandbox,
The button seems to disappear when scrolling to the bottom(around section 5).


I fixed it, 
[see](https://codesandbox.io/s/intersectionobserver-totop-scroll-5t049?file=/src/index.js)

sae-github replied:
Thank you very much.
Fixed.
MERGE.
